bento_federation_service/search/dataset_search/dataset_search.py

```diff
@@ -207,19 +219,47 @@ async def _table_search_worker(
             if not is_querying_data_type:
                 continue
 
+
+            # Setup up search pre-requisites
+            # - defaults:
+            path_fragment=(
+                f"api/{table_ownership['service_artifact']}{'/private' if private else ''}/tables"
+                f"/{table_record['id']}/search"
+            )
+            url_args = (("query", json.dumps(data_type_queries[table_data_type])),)
+
+            # - Gohan compatibility
+            # TODO: formalize/clean this up
+            if USE_GOHAN and table_ownership['service_artifact'] == "variant":
+                # reset path_fragment:
+                path_fragment = (f"api/gohan/variants/get/by/variantId")
+
+                # reset url_args:
+                # - construct based on search query
+                supplemental_url_args = [["getSampleIdsOnly", "true"]]
+                # - transform custom Query to list of lists to simplify
+                #   the gohan query parameter construction
+                tmpjson=json.dumps({"tmpkey":data_type_queries[table_data_type]})
+                reloaded_converted=json.loads(tmpjson)["tmpkey"]
+                # - generate query parameters from list of query tree objects
+                gohan_query_params = query_utils.construct_gohan_query_params(reloaded_converted, supplemental_url_args)
+                url_args = gohan_query_params
+
+
+            # Run the search
             r = await peer_fetch(
                 client,
                 CHORD_URL,
-                path_fragment=(
-                    f"api/{table_ownership['service_artifact']}{'/private' if private else ''}/tables"
-                    f"/{table_record['id']}/search"
-                ),
-                url_args=(("query", json.dumps(data_type_queries[table_data_type])),),
+                path_fragment=path_fragment,
+                url_args=url_args,
                 method="GET",
                 auth_header=auth_header,  # Required in some cases to not get a 403
                 extra_headers=DATASET_SEARCH_HEADERS,
             )
 
+            # if "gohan" in path_fragment:
+            #     print(f"Response: {r}")
```

Comment:
if these arent being used id remove this and other comments below

Reply:
good catch, thanks